In [12]:
import pandas as pd
import numpy as np
import statistics as stat
import random

1. Table 1 shows a frequency distribution of grades on a final examination in
college algebra. Find the quartiles of the distribution.

In [29]:
# grade_class = {'90-100': 9, '80-89': 32, '70-79': 43, '60-69':21, '50-59': 11, '40-49':3, '30-39':1, 'Total': 120}
grade_class = ['30-39', '40-49', '50-59','60-69',  '70-79',  '80-89', '90-99']
freq = [1,3,11, 21,43,32, 9]

freq_df = pd.DataFrame({'No. of Students': freq})
grade_class_df= pd.DataFrame({'Grade': grade_class})
freq_df['Cumulative Freq'] = freq_df['No. of Students'].cumsum()


freq_table = pd.concat([grade_class_df, freq_df], axis=1)
freq_table

Grade  No. of Students  Cumulative Freq
0  30-39                1                1
1  40-49                3                4
2  50-59               11               15
3  60-69               21               36
4  70-79               43               79
5  80-89               32              111
6  90-99                9              120

In [30]:
C = 10
N = 120

Q1 = 80 + (10/32)*(30-9)
Q2 = 70 + (10/43)*(60-41)
Q3 = 60 + (10/21)*(90-84)


Q1 = 60 + (9/21)*(30-15)
Q2 = 70 + (9/43)*(60-36)
Q3 = 80 + (9/32)*(90-79)
quartiles = {'Q1': Q1, 'Q2': Q2, 'Q3': Q3}
quartiles_df = pd.DataFrame(quartiles.items())
quartiles_df

0          1
0  Q1  66.428571
1  Q2  75.023256
2  Q3  83.093750

(5 points). On a final examination in statistics, the mean grade of a group of 150 students was 78 and the standard deviation was 8.0. In algebra, however, the mean final grade of the group was 73 and the standard deviation was 7.6. In which subject was there the greater (a) absolute dispersion and (b) relative dispersion?

In [6]:
n = 150
stat_mean = 78
stat_sd = 8.0
stat_rel_disp = stat_sd/stat_mean

algeb_mean = 73
algeb_sd = 7.6
algeb_rel_disp = algeb_sd/algeb_mean

display(round(stat_rel_disp,4))
display(round(algeb_rel_disp,4))

0.1026

0.1041

a) There is a **greater absolute dispersion in statistics** as the scores have a higher standard deviation of 8. \
b) There is a **greater relative dispersion in algebra** as the scores have a higher relative dispersion of approximately 0.1041.

3. Prove that the mean and standard deviation of a set of standard scores are
equal to 0 and 1, respectively. Use the following problem to illustrate this: Convert the
set 6, 2, 8, 7, 5 into standard scores. 

In [8]:
scores = [6,2,8,7,5]
scores_mean = np.mean(scores)
scores_sd = stat.stdev(scores)  
z_1 = (6 - scores_mean)/scores_sd 
z_2 = (2 - scores_mean)/scores_sd 
z_3 = (8 - scores_mean)/scores_sd 
z_4 = (7 - scores_mean)/scores_sd 
z_5 = (5 - scores_mean)/scores_sd 

standard_mean = np.mean([z_1, z_2, z_3, z_4, z_5])
standard_sd = stat.stdev([z_1, z_2, z_3, z_4, z_5])

display(round(standard_mean,4))
display(round(standard_sd,4))


0.0

1.0

As shown, the **mean of the standard scores is 0** and the **standard deviation of standard scores is 1**. 

4. Three masses are measured as 20.48, 35.97, and 62.34 g, with standard
deviations of 0.21, 0.46, and 0.54 g, respectively. Find the (a) mean and (b) standard
deviation of the sum of the masses

In [10]:
masses = [20.48, 35.97, 62.34] 
masses_sd = [0.21, 0.46, 0.54]

def square(list):
    return [i ** 2 for i in list]

masses_var = square(masses_sd)

mass_sum_mean = np.sum(masses)
mass_sum_sdev = np.sqrt(np.sum(masses_var))


print(f"Mean: {mass_sum_mean} \nSDev:{mass_sum_sdev}")

Mean: 118.79 
SDev:0.7397972695272672


5. Find 𝜇 and 𝜎2 . Give the 25 (with replacement) possible samples of size 2, their means,
and their probabilities

In [11]:
x = [6,9,12,15,18]
px = [0.1,0.2,0.4,0.2,0.1]

def square(list):
    return [i ** 2 for i in list]
x2 = square(x)
mean = 0
n=0

# mean
for i in x:
    mean += x[n]*px[n]
    n+=1
# variance
x2_sum = 0
n = 0
for i in x2:
    x2_sum += x2[n]*px[n]
    n+=1
var = x2_sum - (mean**2)

print(f"Mean: {round(mean,4)}\nVariance: {round(var,4)}")

Mean: 12.0
Variance: 10.8


In [31]:
samples =[]

for i in x:
    for j in x:
        samples.append((i,j))


sample_mean = [(s[0] +s[1])/2 for s in samples]
samples

[(6, 6),
 (6, 9),
 (6, 12),
 (6, 15),
 (6, 18),
 (9, 6),
 (9, 9),
 (9, 12),
 (9, 15),
 (9, 18),
 (12, 6),
 (12, 9),
 (12, 12),
 (12, 15),
 (12, 18),
 (15, 6),
 (15, 9),
 (15, 12),
 (15, 15),
 (15, 18),
 (18, 6),
 (18, 9),
 (18, 12),
 (18, 15),
 (18, 18)]

In [32]:
sample_mean

[6.0,
 7.5,
 9.0,
 10.5,
 12.0,
 7.5,
 9.0,
 10.5,
 12.0,
 13.5,
 9.0,
 10.5,
 12.0,
 13.5,
 15.0,
 10.5,
 12.0,
 13.5,
 15.0,
 16.5,
 12.0,
 13.5,
 15.0,
 16.5,
 18.0]